src/optparse.coffee

```diff
@@ -75,41 +67,98 @@ exports.OptionParser = class OptionParser
 # Helpers
 # -------
 
-# Regex matchers for option flags.
+# Regex matchers for option flags on the command line and their rules.
 LONG_FLAG  = /^(--\w[\w\-]*)/
 SHORT_FLAG = /^(-\w)$/
 MULTI_FLAG = /^-(\w{2,})/
+# Matches the long flag part of a rule for an option with an argument. Not
+# applied to anything in process.argv.
 OPTIONAL   = /\[(\w+(\*?))\]/
 
 # Build and return the list of option rules. If the optional *short-flag* is
 # unspecified, leave it out by padding with `null`.
-buildRules = (rules) ->
-  for tuple in rules
+buildRules = (ruleDecls) ->
+  ruleList = for tuple in ruleDecls
     tuple.unshift null if tuple.length < 3
     buildRule tuple...
+  flagDict = {}
+  for rule in ruleList
+    # shortFlag is null if not provided in the rule.
+    for flag in [rule.shortFlag, rule.longFlag] when flag?
+      if flagDict[flag]?
+        throw new Error "flag #{flag} for switch #{rule.name}
+          was already declared for switch #{flagDict[flag].name}"
```

Comment:
This produces a stack trace. This specific error indicates a bug in the flag declarations given to the `OptionParser` class, which indicates a bug in the program using these -- that might be appropriate to leave as a stack trace. The rest (errors popping up during parsing the given options) should probably be caught and handled in `command.coffee` where it calls `.parse()`. I'll do that.